storage/remote/write_handler.go

```diff
@@ -124,7 +129,13 @@ func (h *writeHandler) write(ctx context.Context, req *prompb.WriteRequest) (err
 		}
 		var ref storage.SeriesRef
 		for _, s := range ts.Samples {
-			ref, err = app.Append(ref, labels, s.Timestamp, s.Value)
+			// check, if the histogram timestamp is in the past
+			if s.Timestamp <= currentTimestamp {
+				ref, err = app.Append(ref, labels, s.Timestamp, s.Value)
+			} else {
+				err = storage.ErrOutOfBounds
+			}
```

Comment:
I think I would do this differently, to avoid logging a confusing message:
```suggestion
			if s.Timestamp > currentTimestamp {
				level.Error(h.logger).Log("msg", "Sample with future timestamp from remote write", "series", labels.String(), "timestamp", s.Timestamp)
				return storage.ErrOutOfBounds
			}
			ref, err = app.Append(ref, labels, s.Timestamp, s.Value)
```